server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalMultiValueAggregation.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.aggregations.metrics;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.InternalAggregation;
+import org.elasticsearch.search.aggregations.support.AggregationPath;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+public abstract class InternalMultiValueAggregation extends InternalAggregation implements MultiValueAggregation {
+
+    private static final DocValueFormat DEFAULT_FORMAT = DocValueFormat.RAW;
+
+    protected DocValueFormat format = DEFAULT_FORMAT;
+
+    protected InternalMultiValueAggregation(String name, Map<String, Object> metadata) {
+        super(name, metadata);
+    }
+
+    /**
+     * Read from a stream.
+     */
+    protected InternalMultiValueAggregation(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    public final double sortValue(AggregationPath.PathElement head, Iterator<AggregationPath.PathElement> tail) {
+        throw new IllegalArgumentException("Metrics aggregations cannot have sub-aggregations (at [>" + head + "]");
+    }
+
+    @Override
+    protected boolean mustReduceOnSingleInternalAgg() {
```

Comment:
I *think* this is something the leaf classes probably should implement rather than here.